### ISSUE TITLE
YOLO10 - RuntimeWarning: invalid value encountered in cast    -  on plot boxes 

### ISSUE THREAD

**Issue by idubi-google-reichman** [ultralytics/ultralytics#14906]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### when executing yolo8 did not get his issue, only yolo10
executing same data on yolo10 getting this warning : 
```
plotting.py:1025: RuntimeWarning: invalid value encountered in cast
  for j, box in enumerate(boxes.astype(np.int64).tolist()):
```

it appear to happen when some boxes has only nan values :

**python line execution :** 
```
    model = YOLO("yolov10n.yaml")
    model.train(data="./resources/dataset/data.yaml", epochs=2)

```

```

                Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:01<00:00,  1.45it/s]
                   all         45        189          0          0          0          0
 =====================================================  
  boxes = [[     265.36      88.929       247.5      143.57]] 
 =====================================================
 =====================================================
  boxes = [[     269.64         780      127.14        87.5]
 [      177.5      814.29      94.286      70.714]]
 =====================================================
 =====================================================  
  boxes = [[     285.36      226.79      88.929      116.79]]
 =====================================================
 =====================================================
 =====================================================
  boxes = [[        nan         nan         nan         nan]
 [        nan         nan         nan         nan]
 [        nan         nan         nan         nan]
 ...
 [        nan         nan         nan         nan]
 [        nan         nan         nan         nan]
 [        nan         nan         nan         nan]]
 =====================================================
 =====================================================
  boxes = [[        nan         nan         nan         nan]
 [        nan         nan         nan         nan]
 [        nan         nan         nan         nan]
 ...
 [        nan         nan         nan         nan]
 [        nan         nan         nan         nan]
 [        nan         nan         nan         nan]]
 =====================================================
  boxes = [[     269.64        1260      127.14        87.5]
 [      177.5      1294.3      94.286      70.714]]   boxes = [[     300.36      680.71      71.429      88.571]
 [     174.64      691.79      61.786      81.429]]

 =====================================================
 =====================================================
D:\projects\AI\deep-learning-class\project2\coins\.venv-coins\lib\site-packages\ultralytics\utils\plotting.py:1025: RuntimeWarning: invalid value encountered in cast
  for j, box in enumerate(boxes.astype(np.int64).tolist()):
 =====================================================
 =====================================================
  boxes = [[     300.36      1160.7      71.429      88.571]
 [     174.64      1171.8      61.786      81.429]]
 =====================================================
 =====================================================  
 =====================================================
 =====================================================
  boxes = [[        nan         nan         nan         nan]
 [        nan         nan         nan         nan]
 [        nan         nan         nan         nan]
 ...
 [        nan         nan         nan         nan]
 [        nan         nan         nan         nan]
 [        nan         nan         nan         nan]]   boxes = [[        nan         nan         nan         nan]
 [        nan         nan         nan         nan]
 [        nan         nan         nan         nan]
 ...
 [        nan         nan         nan         nan]
 [        nan         nan         nan         nan]
 [        nan         nan         nan         nan]]
 =====================================================

 =====================================================
 =====================================================
  boxes = [[      222.5      1780.4      31.071      43.571]
 [     190.36      1703.9      32.143      40.714]
 [     139.29      1762.5      33.929      41.429]
 [        165      1861.8      23.929      37.143]
 [     197.14      1849.3        27.5        32.5]
 [     276.43      1767.1      25.357      37.857]
 [     212.14      1746.1      27.143      34.286]
 [     184.29      1749.3      27.857      34.286]
 [     240.71      1596.8      22.857          35]
 [     295.71      1618.2      22.857      36.786]
 [     327.14      1580.4      21.786      36.786]]
  boxes = [[     153.21      1667.9      66.071        82.5]]
 =====================================================  
 =====================================================
  boxes = [[        nan         nan         nan         nan]
 [        nan         nan         nan         nan]
 [        nan         nan         nan         nan]
 ...
 [        nan         nan         nan         nan]
 [        nan         nan         nan         nan]
 [        nan         nan         nan         nan]]
 =====================================================
 



```

### Environment

`$ yolo checks`
```
Ultralytics YOLOv8.2.67 🚀 Python-3.10.11 torch-2.5.0.dev20240726+cu124 CUDA:0 (Quadro T1000, 4096MiB)
Setup complete ✅ (12 CPUs, 63.7 GB RAM, 160.5/237.2 GB disk)

OS                  Windows-10-10.0.19045-SP0
Environment         Windows
Python              3.10.11
Install             git
RAM                 63.74 GB
CPU                 Intel Core(TM) i7-9850H 2.60GHz
CUDA                12.4

numpy               ✅ 1.26.4<2.0.0,>=1.23.0
matplotlib          ✅ 3.9.1>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 9.3.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.14.0>=1.4.1
torch               ✅ 2.5.0.dev20240726+cu124>=1.8.0
torchvision         ✅ 0.20.0.dev20240727+cu124>=0.9.0
tqdm                ✅ 4.66.4>=4.64.0
psutil              ✅ 6.0.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.2>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.0>=2.0.0
(.venv-coins) 

```
### Minimal Reproducible Example

cuda is available
```
New https://pypi.org/project/ultralytics/8.2.71 available 😃 Update with 'pip install -U ultralytics'
Ultralytics YOLOv8.2.67 🚀 Python-3.10.11 torch-2.5.0.dev20240726+cu124 CUDA:0 (Quadro T1000, 4096MiB)
engine\trainer: task=detect, mode=train, model=yolov10n.yaml, data=./resources/dataset/data.yaml, epochs=2, time=None, patience=100, batch=16, imgsz=640, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=train60, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=1.0, mixup=0.0, copy_paste=0.0, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=D:\projects\AI\deep-learning-class\project2\runs\detect\train60
Overriding model.yaml nc=80 with nc=4

                   from  n    params  module                                       arguments
  0                  -1  1       464  ultralytics.nn.modules.conv.Conv             [3, 16, 3, 2]
  1                  -1  1      4672  ultralytics.nn.modules.conv.Conv             [16, 32, 3, 2]
  2                  -1  1      7360  ultralytics.nn.modules.block.C2f             [32, 32, 1, True]
  3                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]
  4                  -1  2     49664  ultralytics.nn.modules.block.C2f             [64, 64, 2, True]
  5                  -1  1      9856  ultralytics.nn.modules.block.SCDown          [64, 128, 3, 2]
  6                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]
  7                  -1  1     36096  ultralytics.nn.modules.block.SCDown          [128, 256, 3, 2]
  8                  -1  1    460288  ultralytics.nn.modules.block.C2f             [256, 256, 1, True]
  9                  -1  1    164608  ultralytics.nn.modules.block.SPPF            [256, 256, 5]
 10                  -1  1    249728  ultralytics.nn.modules.block.PSA             [256, 256]
 11                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 12             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 13                  -1  1    148224  ultralytics.nn.modules.block.C2f             [384, 128, 1]
 14                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 15             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 16                  -1  1     37248  ultralytics.nn.modules.block.C2f             [192, 64, 1]
 17                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]
 18            [-1, 13]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 19                  -1  1    123648  ultralytics.nn.modules.block.C2f             [192, 128, 1]
 20                  -1  1     18048  ultralytics.nn.modules.block.SCDown          [128, 128, 3, 2]
 21            [-1, 10]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 22                  -1  1    282624  ultralytics.nn.modules.block.C2fCIB          [384, 256, 1, True, True]     
 23        [16, 19, 22]  1    862888  ultralytics.nn.modules.head.v10Detect        [4, [64, 128, 256]]
YOLOv10n summary: 385 layers, 2,708,600 parameters, 2,708,584 gradients, 8.4 GFLOPs

Freezing layer 'model.23.dfl.conv.weight'
AMP: running Automatic Mixed Precision (AMP) checks with YOLOv8n...
AMP: checks passed ✅
D:\projects\AI\deep-learning-class\project2\coins\.venv-coins\lib\site-packages\ultralytics\engine\trainer.py:268: FutureWarning: `torch.cuda.amp.GradScaler(args...)` is deprecated. Please use `torch.amp.GradScaler('cuda', args...)` instead.
  self.scaler = torch.cuda.amp.GradScaler(enabled=self.amp)
train: Scanning D:\projects\AI\deep-learning-class\project2\coins\resources\dataset\train\labels... 195 images, 0 backgrounds, 0 corrupt: 100%|██████████| 195/195 [00:00<00:00, 1981.20it/s]
train: New cache created: D:\projects\AI\deep-learning-class\project2\coins\resources\dataset\train\labels.cache
val: Scanning D:\projects\AI\deep-learning-class\project2\coins\resources\dataset\valid\labels... 45 images, 0 backgrounds, 0 corrupt: 100%|██████████| 45/45 [00:00<00:00, 1151.14it/s]
val: New cache created: D:\projects\AI\deep-learning-class\project2\coins\resources\dataset\valid\labels.cache
Plotting labels to D:\projects\AI\deep-learning-class\project2\runs\detect\train60\labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.00125, momentum=0.9) with parameter groups 95 weight(decay=0.0), 108 weight(decay=0.0005), 107 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 8 dataloader workers
Logging results to D:\projects\AI\deep-learning-class\project2\runs\detect\train60
Starting training for 2 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
        1/2      3.17G        nan        nan        nan         11        640: 100%|██████████| 13/13 [00:19<00:00,  1.48s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:02<00:00,  1.32s/it]
                   all         45        218          0          0          0          0

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
        2/2      3.13G        nan        nan        nan         54        640: 100%|██████████| 13/13 [00:18<00:00,  1.43s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:02<00:00,  1.32s/it]
                   all         45        218          0          0          0          0

2 epochs completed in 0.012 hours.
Optimizer stripped from D:\projects\AI\deep-learning-class\project2\runs\detect\train60\weights\last.pt, 5.7MB
Optimizer stripped from D:\projects\AI\deep-learning-class\project2\runs\detect\train60\weights\best.pt, 5.7MB

Validating D:\projects\AI\deep-learning-class\project2\runs\detect\train60\weights\best.pt...
Ultralytics YOLOv8.2.67 🚀 Python-3.10.11 torch-2.5.0.dev20240726+cu124 CUDA:0 (Quadro T1000, 4096MiB)
YOLOv10n summary (fused): 285 layers, 2,695,976 parameters, 0 gradients, 8.2 GFLOPs
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:01<00:00,  1.41it/s]
                   all         45        218          0          0          0          0
**D:\projects\AI\deep-learning-class\project2\coins\.venv-coins\lib\site-packages\ultralytics\utils\plotting.py:1022: RuntimeWarning: invalid value encountered in cast**
  for j, box in enumerate(boxes.astype(np.int64).tolist()):
Speed: 0.3ms preprocess, 26.7ms inference, 0.0ms loss, 0.4ms postprocess per image
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by ambitious-octopus** [comment]
Hey @idubi-google-reichman, your code is not reproducible because we have no information about your data. It could be a mistake in the labels of your dataset. Have you tried using one of our datasets? Find the list [here](https://docs.ultralytics.com/datasets/).

**Comment by Y-T-G** [comment]
Could be a torch issue. Downgrade to 2.3.1.

**Comment by Y-T-G** [comment]
I noticed you're using Quadro T1000. It has issue with FP16. You can try predicting with `half=False`.
